drracket/drracket/private/language.rkt

```diff
@@ -746,47 +740,37 @@
                                                    init-code
                                                    mred-launcher
                                                    use-copy?)
-    (let ([executable-specs (create-executable-gui parent
-                                                   program-filename
-                                                   #t 
-                                                   (if (boolean? mred-launcher)
-                                                       (if mred-launcher
-                                                           'mred
-                                                           'mzscheme)
-                                                       #t))])
-      (when executable-specs
-        (let* ([type (car executable-specs)]
-               [base (cadr executable-specs)]
-               [executable-filename (caddr executable-specs)]
-               [aux (cadddr executable-specs)]
-               [create-executable
-                (case type
-                  [(launcher) create-module-based-launcher]
-                  [(stand-alone) create-module-based-stand-alone-executable]
-                  [(distribution) create-module-based-distribution])])
-          (with-handlers ((exn:fail? (λ (msg)
-                                       (define sp (open-output-string))
-                                       (parameterize ([current-error-port sp])
-                                         (drracket:init:original-error-display-handler
-                                          (exn-message exn)
-                                          exn))
-                                       (message-box 
-                                        (string-constant drscheme)
-                                        (string-append
-                                         (string-constant error-creating-executable)
-                                         "\n\n"
-                                         (get-output-string sp))))))
-            (create-executable
-             program-filename
-             executable-filename
-             module-language-spec
-             transformer-module-language-spec
-             init-code
-             (if (boolean? mred-launcher)
-                 mred-launcher
-                 (eq? base 'mred))
-             use-copy?
-             #:aux aux))))))
+    (define executable-specs
+      (create-executable-gui parent
+                             program-filename
+                             #t
+                             (if (boolean? mred-launcher) (if mred-launcher 'mred 'mzscheme) #t)))
+    (when executable-specs
```

Comment:
Here too I think I had the newlines (especially for the last true) intentionally. This doesn't seem to matter too much, but the ones above add a lot of rightward drift.

Reply:
This can also be rewritten as a `cond`:

```scheme
(cond
  [(not (boolean? mred-launcher)) #t]
  [mred-launcher 'mred]
  [else 'mzscheme])
```

Resyntax doesn't currently have a rule for that, but it could. Do you think this would be better?

Reply:
I think I prefer the nested one but it isn't a strong preference.

Reply:
@sorawee What do you think about not allowing `if` to be a single line in `fmt` if either of the branches is also an `if` expression? `(if (a) (if (b) (X) (Y)) (z))` is pretty hard to read even when the subexpressions are short.